Realm.Shared/linq/RealmResultsVisitor.cs

```diff
@@ -389,41 +389,40 @@ private static void AddQueryEqual(QueryHandle queryHandle, string columnName, ob
 
         private static void AddQueryNotEqual(QueryHandle queryHandle, string columnName, object value)
         {
-            var columnIndex = NativeQuery.get_column_index((QueryHandle)queryHandle, columnName, (IntPtr)columnName.Length);
+            var columnIndex = queryHandle.GetColumnIndex(columnName);
 
             var valueType = value.GetType();
```

Comment:
This `GetType` could be moved underneath the string test because it's only used for the other cases